### DISCUSSION TITLE
Can shortcuts be made runnable only when on a specifc wifi?

### DISCUSSION THREAD

**Discussion by FlyveHest** [Waboodoo/HTTP-Shortcuts#455] [General]
Is it possible to make some shortcuts only runnable when on a specific wifi?

For instance, I have some things only available on my LAN, and it would be nice if the shortcuts just didn't run at all if I start them when not on wifi or on a foreign one.

**Comment by Waboodoo** [comment]
Hi,
Yes, this is possible. There are 2 ways to achieve this.

When editing your shortcut, scroll down and click on "Advanced Technical Settings". There, tick the "Require specific Wi-Fi" checkbox and enter the name of the Wi-Fi. Now, whenever you trigger this shortcut and your device is not connected to that Wi-Fi, it will prompt you to connect to the desired Wi-Fi.

If you don't want this prompt and instead want the shortcut to just do nothing, then you'll have to use the Scripting feature instead. In the shortcut editor click on "Scripting", then in the first text box enter the following code:

```js
if (getWifiSSID() != 'name of your wifi here') {
  abort();
}
```

Hope this helps.

**Comment by FlyveHest** [comment]
Absolutely helps, yes, thanks for the quick reply.
